### ISSUE TITLE
Cannot display widget tree in devtool

### ISSUE THREAD

**Issue by yellowgraysu** [suragch/audio_video_progress_bar#46]
### Description
If I add the widget inside my app(web app), the widget inpector won't show the tree structure.

### Step
1. add ProgressBar(progress: Duration(seconds: 20), total: Duration(seconds: 60)) in app
2. run debug build
3. open devTool widget inspector page

<details>
  <summary>error message</summary>

```

════════ Exception caught by Flutter framework ═════════════════════════════════
The following assertion was thrown during a service extension callback for "ext.flutter.inspector.getRootWidgetSummaryTreeWithPreviews":
Assertion failed:
../…/foundation/diagnostics.dart:2137
ifTrue != null || ifFalse != null
is not true


When the exception was thrown, this was the stack
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 266:49      throw_
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 29:3        assertFailed
packages/flutter/src/foundation/diagnostics.dart 2137:40                          new
packages/audio_video_progress_bar/audio_video_progress_bar.dart 338:20            debugFillProperties
packages/flutter/src/widgets/framework.dart 4466:14                               debugFillProperties
packages/flutter/src/widgets/framework.dart 6201:11                               debugFillProperties
packages/flutter/src/foundation/diagnostics.dart 2940:17                          <fn>
packages/flutter/src/foundation/diagnostics.dart 2942:16                          get builder
packages/flutter/src/foundation/diagnostics.dart 2954:77                          get emptyBodyDescription
packages/flutter/src/foundation/diagnostics.dart 1632:13                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1657:21                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1660:14                          toJsonMap
packages/flutter/src/widgets/framework.dart 4632:45                               toJsonMap
packages/flutter/src/foundation/diagnostics.dart 1686:18                          <fn>
dart-sdk/lib/internal/iterable.dart 413:31                                        elementAt
dart-sdk/lib/internal/iterable.dart 342:26                                        moveNext
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 748:20  next
dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 586:14                of
dart-sdk/lib/internal/iterable.dart 213:44                                        toList
packages/flutter/src/foundation/diagnostics.dart 1686:58                          toJsonList
packages/flutter/src/foundation/diagnostics.dart 1654:23                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1657:21                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1660:14                          toJsonMap
packages/flutter/src/widgets/framework.dart 4632:45                               toJsonMap
packages/flutter/src/foundation/diagnostics.dart 1686:18                          <fn>
dart-sdk/lib/internal/iterable.dart 413:31                                        elementAt
dart-sdk/lib/internal/iterable.dart 342:26                                        moveNext
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 748:20  next
dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 586:14                of
dart-sdk/lib/internal/iterable.dart 213:44                                        toList
packages/flutter/src/foundation/diagnostics.dart 1686:58                          toJsonList
packages/flutter/src/foundation/diagnostics.dart 1654:23                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1657:21                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1660:14                          toJsonMap
packages/flutter/src/widgets/framework.dart 4632:45                               toJsonMap
packages/flutter/src/foundation/diagnostics.dart 1686:18                          <fn>
dart-sdk/lib/internal/iterable.dart 413:31                                        elementAt
dart-sdk/lib/internal/iterable.dart 342:26                                        moveNext
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 748:20  next
dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 586:14                of
dart-sdk/lib/internal/iterable.dart 213:44                                        toList
packages/flutter/src/foundation/diagnostics.dart 1686:58                          toJsonList
packages/flutter/src/foundation/diagnostics.dart 1654:23                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1657:21                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1660:14                          toJsonMap
packages/flutter/src/widgets/framework.dart 4632:45                               toJsonMap
packages/flutter/src/foundation/diagnostics.dart 1686:18                          <fn>
dart-sdk/lib/internal/iterable.dart 413:31                                        elementAt
dart-sdk/lib/internal/iterable.dart 342:26                                        moveNext
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/operations.dart 748:20  next
dart-sdk/lib/_internal/js_dev_runtime/patch/core_patch.dart 586:14                of
dart-sdk/lib/internal/iterable.dart 213:44                                        toList
packages/flutter/src/foundation/diagnostics.dart 1686:58                          toJsonList
packages/flutter/src/foundation/diagnostics.dart 1654:23                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1657:21                          <fn>
packages/flutter/src/foundation/diagnostics.dart 1660:14                          toJsonMap
packages/flutter/src/widgets/framework.dart 4632:45                               toJsonMap
packages/flutter/src/widgets/widget_inspector.dart 1606:18                        [_nodeToJson]
%3Canonymous%3E 69:49                                                             getRootWidgetSummaryTreeWithPreviews
packages/flutter/src/foundation/binding.dart 855:32                               <fn>
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50                <fn>
dart-sdk/lib/async/zone.dart 1653:54                                              runUnary
dart-sdk/lib/async/future_impl.dart 147:18                                        handleValue
dart-sdk/lib/async/future_impl.dart 766:44                                        handleValueCallback
dart-sdk/lib/async/future_impl.dart 795:13                                        _propagateToListeners
dart-sdk/lib/async/future_impl.dart 557:7                                         [_complete]
dart-sdk/lib/async/future.dart 421:15                                             <fn>
dart-sdk/lib/_internal/js_dev_runtime/private/isolate_helper.dart 48:19           internalCallback
════════════════════════════════════════════════════════════════════════════════
```
</details>

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 3.3.2, on macOS 12.5.1 21G83 darwin-arm, locale zh-Hant-TW)
    • Flutter version 3.3.2 on channel stable at /Users/grayhuang/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e3c29ec00c (6 days ago), 2022-09-14 08:46:55 -0500
    • Engine revision a4ff2c53d8
    • Dart version 2.18.1
    • DevTools version 2.15.0

[✗] Android toolchain - develop for Android devices
    ✗ Unable to locate Android SDK.
      Install Android Studio from: https://developer.android.com/studio/index.html
      On first launch it will assist you in installing the Android SDK components.
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).
      If the Android SDK has been installed to a custom location, please use
      `flutter config --android-sdk` to update to that location.


[✗] Xcode - develop for iOS and macOS
    ✗ Xcode installation is incomplete; a full installation is necessary for iOS development.
      Download at: https://developer.apple.com/xcode/download/
      Or install Xcode via the App Store.
      Once installed, run:
        sudo xcode-select --switch /Applications/Xcode.app/Contents/Developer
        sudo xcodebuild -runFirstLaunch
    ✗ CocoaPods not installed.
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To install see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/macos#android-setup for detailed instructions).

[✓] VS Code (version 1.71.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.5.1 21G83 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 3 categories.
  ```
</details>

**Comment by thesensibledev** [comment]
Yes. This is replicable and needs a fix. Exception is caused on

```dart
properties.add(FlagProperty('thumbCanPaintOutsideBar',
        value: thumbCanPaintOutsideBar));
```
A fix is to replace is with 
```dart
properties.add(
      FlagProperty(
        'thumbCanPaintOutsideBar',
        value: thumbCanPaintOutsideBar,
        ifTrue: 'true',
        ifFalse: 'false',
        showName: true,
      ),
    );
```
if @suragch doesn't pick it up, I will push a PR.

**Comment by suragch** [comment]
@thesensibledev Thanks for finding the solution so quickly! That would be great if you could submit a PR, but let's finish solving issue #44 first.

**Comment by suragch** [comment]
Thanks to @thesensibledev this is fixed now in version 0.11.0.
